Client/Frontend/Settings/SettingsTableViewController.swift

```diff
@@ -404,14 +404,13 @@ class AccountSetting: Setting, FxAContentViewControllerDelegate {
     override var accessoryType: UITableViewCellAccessoryType { return .none }
 
     func contentViewControllerDidSignIn(_ viewController: FxAContentViewController, withFlags flags: FxALoginFlags) {
-        // Reload the data to reflect the new Account immediately.
-        settings.tableView.reloadData()
-        // And start advancing the Account state in the background as well.
-        settings.SELrefresh()
-
         // Dismiss the FxA content view if the account is verified.
         if flags.verified {
             _ = settings.navigationController?.popToRootViewController(animated: true)
+            // Reload the data to reflect the new Account immediately.
+            settings.tableView.reloadData()
+            // And start advancing the Account state in the background as well.
+            settings.SELrefresh()
```

Comment:
I don't understand how this fixes the issue. Wouldn't the previous way just *always* call this instead of conditionally checking `flags.verified`?

Reply:
Previous behaviour was that this method was being called multiple times; and that the settings screen may be being displayed, but the "Sync Now" was either displayed as "Sync Now" as if the sync wasn't happening. 

This was because the sync correctly, _wasn't happening_, but the FxA webview was being dismissed too early.

The delegate method is still being called multiple times, but the flags are being respected – i.e. the webview is only being dismissed once the account is verified.

Reply:
Ah ok. Maybe we should include your explanation in a comment?

Reply:
Updated with comment.